service/kernel/data_management/train_test_split.cpp

```diff
@@ -0,0 +1,253 @@
+/** file train_test_split.cpp */
+/*******************************************************************************
+* Copyright 2020 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "data_management/data/internal/train_test_split.h"
+#include "data_management/data/numeric_table.h"
+#include "data_management/data/data_dictionary.h"
+#include "services/env_detect.h"
+#include "externals/service_dispatch.h"
+#include "service/kernel/service_data_utils.h"
+#include "service/kernel/service_utils.h"
+#include "service/kernel/service_defines.h"
+#include "algorithms/threading/threading.h"
+#include "service/kernel/data_management/service_numeric_table.h"
+#include "data_management/features/defines.h"
+
+namespace daal
+{
+namespace data_management
+{
+namespace internal
+{
+typedef daal::data_management::NumericTable::StorageLayout NTLayout;
+
+const size_t BLOCK_CONST      = 2048;
+const size_t THREADING_BORDER = 8388608;
+
+template <typename Func>
+void runBlocks(const bool inParallel, const size_t nBlocks, Func func)
+{
+    if (inParallel)
+        daal::threader_for(nBlocks, nBlocks, [&](size_t i) { func(i); });
+    else
+        for (size_t i = 0; i < nBlocks; ++i) func(i);
+}
+
+template <typename DataType, typename IdxType>
+services::Status assignColumnValues(const DataType * origDataPtr, const NumericTablePtr dataTable, const IdxType * idxPtr, const size_t startRow,
+                                    const size_t nRows, const size_t iCol)
+{
+    services::Status s;
+
+    BlockDescriptor<DataType> dataBlock;
+    dataTable->getBlockOfColumnValues(iCol, startRow, nRows, readWrite, dataBlock);
+    DataType * dataPtr = dataBlock.getBlockPtr();
+    DAAL_CHECK_MALLOC(dataPtr);
+
+    for (size_t i = 0; i < nRows; ++i) dataPtr[i] = origDataPtr[idxPtr[i]];
```

Comment:
I suppose by coding guidelines it should be:
```cpp
for (size_t i = 0; i < nRows; ++i)
{
    dataPtr[i] = origDataPtr[idxPtr[i]];
}
```

Reply:
Fixed